contracts/ColonyStorage.sol

```diff
@@ -206,9 +206,11 @@ contract ColonyStorage is CommonStorage, ColonyDataTypes, ColonyNetworkDataTypes
     require(domainExists(_permissionDomainId), "ds-auth-permission-domain-does-not-exist");
     require(domainExists(_childDomainId), "ds-auth-child-domain-does-not-exist");
     require(isAuthorized(msg.sender, _permissionDomainId, msg.sig), "ds-auth-unauthorized");
-    if (_permissionDomainId != _childDomainId) {
-      require(validateDomainInheritance(_permissionDomainId, _childSkillIndex, _childDomainId), "ds-auth-invalid-domain-inheritence");
-    }
```

Comment:
Fair, but alternatively I think it's time to bite the bullet and just add the if/else to `validateDomainInheritance`. Otherwise the `permission == domain` logic keeps leaking out all over the rest of the codebase for no reason.

Reply:
That works for me too and is better.